test/Tests/src/Data/Vector_Spec.enso

```diff
@@ -292,16 +291,39 @@ type_spec name alter = Test.group name <|
         mixed.filter Filter_Condition.Is_Empty . should_equal [Nothing]
         mixed.filter Filter_Condition.Not_Empty . should_equal [1, "b"]
 
-        boolvec = [True, False, Nothing, True]
-        boolvec.filter Filter_Condition.Is_True . should_equal [True, True]
-        boolvec.filter Filter_Condition.Is_False . should_equal [False]
-
-        numvec = [1, 2.5, Number.nan, Number.positive_infinity, Number.negative_infinity, 0]
+        numvec = alter [1, 2.5, Number.nan, Number.positive_infinity, Number.negative_infinity, 0]
         # We need to use to_text because NaN!=NaN
         numvec.filter Filter_Condition.Is_Nan . map .to_text . should_equal ["NaN"]
         numvec.filter Filter_Condition.Is_Infinite . should_equal [Number.positive_infinity, Number.negative_infinity]
         numvec.filter Filter_Condition.Is_Finite . should_equal [1, 2.5, 0]
 
+        Test.expect_panic Type_Error (txtvec.filter Filter_Condition.Is_Finite)
+
+        (alter [2, "a"]).filter (Filter_Condition.Greater 1) . should_fail_with Incomparable_Values
+
+    Test.specify "should allow Nothing when filtering by Filter_Condition" <|
+        (alter [1, 2, Nothing, 3]).filter (Filter_Condition.Greater 2) . should_equal [3]
+        (alter [1, 2, Nothing, 3]).filter (Filter_Condition.Equal_Or_Less 2) . should_equal [1, 2]
+        (alter ["a", 2, Nothing, 2]).filter (Filter_Condition.Equal 2) . should_equal [2, 2]
+        (alter ["a", 2, Nothing, "a", "a"]).filter (Filter_Condition.Equal "a") . should_equal ["a", "a", "a"]
+
+        (alter [1, Nothing, (1/0), (0.log 0)]).filter Filter_Condition.Is_Nan . map .to_text . should_equal ["NaN"]
+        (alter [1, Nothing, (1/0), (0.log 0)]).filter Filter_Condition.Is_Infinite . should_equal [Number.positive_infinity]
+        (alter [1, Nothing, (1/0), (0.log 0)]).filter Filter_Condition.Is_Finite . should_equal [1]
+
+        boolvec = alter [True, False, Nothing, True]
+        boolvec.filter Filter_Condition.Is_True . should_equal [True, True]
+        boolvec.filter Filter_Condition.Is_False . should_equal [False]
+
+        txtvec = alter ["abab", "baaa", Nothing, "cccc", "BAAA"]
+        txtvec.filter (Filter_Condition.Equal_Ignore_Case "baaA") . should_equal ["baaa", "BAAA"]
+        txtvec.filter (Filter_Condition.Contains "a") . should_equal ["abab", "baaa"]
+        txtvec.filter (Filter_Condition.Starts_With "a") . should_equal ["abab"]
+        txtvec.filter (Filter_Condition.Ends_With "a") . should_equal ["baaa"]
+        txtvec.filter (Filter_Condition.Like "b%a") . should_equal ["baaa"]
+        # Nothing is not included in the negation either
+        txtvec.filter (Filter_Condition.Not_Like "b%a") . should_equal ["abab", "cccc", "BAAA"]
```

Comment:
Note, here we 'break' the property that
```
forall v p,
(v.filter (Filter_Condition.Like p)) + (v.filter (Filter_Condition.Not_Like p)) . length == v.length
```

because the `Nothing` values do not get into either category.

I think that is _OK_, because `Like` and `Not_Like` are both _Text_ filters that return _Text_s that satisfy a condition, `Nothing` is neither like nor not-like a pattern.

Do you think that is OK?

I'm also fine with including `Nothing` in the `Not_Like` case if that is preferred.